infra-server&backend

Js-o opened the pull request:
Database and server up and running

Model and controllers need further refinement

Commit:
Cleaning up files for merge with upstream

Deleted server-dist

Commit:
moved files to src

from inside server

Commit:
Minor refactoring in models

Renames the query to controller
moved db connection into server.js for testing purposes

Commit:
Connected DB and server

Commit:
Removed db connect

Not needed in server.js

billytianlan commented on package.json:
```diff
@@ -40,7 +40,9 @@
     "react": "^15.3.0",
-    "react-dom": "^15.3.0"
+    "react-dom": "^15.3.0",
+    "pg-hstore": "^2.3.2",
```
We should remove pg-hstore this for now

billytianlan commented on server/src/Profile/ProfileController.js:
```diff
@@ -0,0 +1,63 @@
+};
+
+let addProfile = (req, res) => {
+  Profile.count({
```
What is the .count method?

billytianlan commented on server/src/Users/UserModel.js:
```diff
@@ -0,0 +1,39 @@
+import db from '../db/db-config';
+import Sequelize from 'Sequelize';
```
lowercase the Sequelize module

pranaymohan commented on server/src/db/db-config.js:
```diff
@@ -0,0 +1,41 @@
+    protocol: 'postgres',
+    dialect: 'postgres'
+
+    , pool: {
```
syntactical issue -- weird spacing on github, is it the same locally?

pranaymohan commented on server/src/db/db-config.js:
```diff
@@ -0,0 +1,41 @@
+      idle: 10000
+    }
+  })
+//and dev environment
```
Instead of using an if/else block here, can we define the host using a ternary statement checking the value of process.env.DATABASE_URL?

Eg. host = process.env.DATABASE_URL ? process.env.DATABASE_URL.split(':')[2] : 'localhost';

EDIT - You can also configure environmental variables to check for this globally, so you don't have to keep replacing urls with 'localhost' in the app. We can probably push this refinement to later though.

pranaymohan commented on server/src/routes/router.js:
```diff
@@ -0,0 +1,19 @@
+import user from '../Users/UserController';
```
We should be more specific about our imported variables when we have both controller and model. Elsewhere, the user model is imported as 'User'. Here, maybe we should import as 'userController'?

pranaymohan commented on server/src/Users/UserController.js:
```diff
@@ -0,0 +1,84 @@
+  .catch(err => {
+    console.log('Error: ', err);
+    done(err);
+  })
```
missing semicolon :)

pranaymohan commented on server/src/Profile/ProfileController.js:
```diff
@@ -0,0 +1,63 @@
+    .then(profile => {
+      console.log('Created new profile!');
+      res.end();
+    })
```
needs a catch block I think

Commit:
Updated based on comments

1. Deleted hstore from package.json
2. Lower case for sequelize
3. syntax for comma and pool
4. iternary statement for if/else in router
